sphinx/texinputs/sphinx.sty

```diff
@@ -361,6 +363,28 @@
 \newcommand\sphinxsetup[1]{\setkeys{sphinx}{#1}}
 
 
+%% CYRILLIC IN NON-CYRILLIC DOCUMENTS (pdflatex only)
+%
+% refs: https://tex.stackexchange.com/q/460271/4686
+%
+\ifspx@opt@cyrnocyr
+  \@tfor\@tempa:=%
+    {ae}{a}{b}{chrdsc}{chvcrs}{ch}{c}{dje}{dze}{dzhe}{d}{erev}{ery}{e}%
+    {f}{ghcrs}{gup}{g}{hdsc}{hrdsn}{h}{ie}{ii}{ishrt}{i}{je}%
+    {kbeak}{kdsc}{kvcrs}{k}{lje}{l}{m}{ndsc}{ng}{nje}{n}{otld}{o}{p}{r}%
+    {schwa}{sdsc}{sftsn}{shch}{shha}{sh}{s}{tshe}{t}{ushrt}{u}{v}%
+    {ya}{yhcrs}{yi}{yo}{yu}{y}{zdsc}{zhdsc}{zh}{z}\do
+  {%
+    \expandafter\DeclareTextSymbolDefault\expandafter
+               {\csname cyr\@tempa\endcsname}{T2A}%
+    \expandafter\uppercase\expandafter{\expandafter
+                 \def\expandafter\@tempa\expandafter{\@tempa}}%
+    \expandafter\DeclareTextSymbolDefault\expandafter
+      {\csname CYR\@tempa\endcsname}{T2A}%
+  }%
+  \DeclareTextSymbolDefault{\CYRpalochka}{T2A}%
+\fi
+
```

Comment:
this fancy LaTeX code is what is needed for LaTeX to automatically use T2A font encoding when encountering a Cyrillic letter.